pkg/auth/handler/oauth/userinfo.go

```diff
@@ -4,20 +4,19 @@ import (
 	"encoding/json"
 	"net/http"
 
-	"github.com/gorilla/mux"
 	"github.com/lestrrat-go/jwx/jwt"
 
 	"github.com/skygeario/skygear-server/pkg/auth/dependency/auth"
 	"github.com/skygeario/skygear-server/pkg/auth/dependency/oauth"
 	"github.com/skygeario/skygear-server/pkg/db"
+	"github.com/skygeario/skygear-server/pkg/httproute"
 	"github.com/skygeario/skygear-server/pkg/log"
 )
 
-func ConfigureUserInfoHandler(router *mux.Router, h http.Handler) {
-	router.NewRoute().
-		Path("/oauth2/userinfo").
-		Methods("GET", "POST", "OPTIONS").
-		Handler(oauth.RequireScope(h))
+func ConfigureUserInfoHandler(route httproute.Route, h http.Handler) (httproute.Route, http.Handler) {
+	return route.
+		WithMethods("GET", "POST", "OPTIONS").
+		WithPathPattern("/oauth2/userinfo"), oauth.RequireScope(h)
```

Comment:
`oauth.RequireScope` is actually a middleware. Is it better to refactor its application to main package to make all `ConfigureHandler` functions have same signature?

Reply:
The only usage is here and `scopes` is actually empty. So `RequireScope` does not have any effect? Am I correct?

Reply:
Sorry for the bad name. `RequireScope` requires two things:
- a valid access token; and
- the access token has the specified scopes (if provided).

So here it just requires a valid access token, without checking for scopes.